src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -1,5 +1,67 @@
 package org.skypro.skyshop.basket;
 
+import org.skypro.skyshop.product.Product;
+
+import java.awt.desktop.AppReopenedEvent;
+import java.net.SocketTimeoutException;
+import java.util.Arrays;
+
 public class ProductBasket {
+    public Product[] array;
+    private int productsCount = 0;
+
+    public ProductBasket() {
+        this.array = new Product[5];
+    }
+
+    public void addProductToBasket(Product product) {
+        if (this.productsCount > 4) {
```

Comment:
используем именно длину массива
```suggestion
        if (this.productsCount > array.length - 1) {
```